pyphare/pyphare/pharesee/hierarchy.py

```diff
@@ -1061,6 +1378,109 @@ def dist_plot(self, **kwargs):
         return final, dp(final, **kwargs)
 
 
+
+
+class ScalarField(PatchHierarchy):
+
+    def __init__(self, patch_levels, domain_box, refinement_ratio=2, time=0., data_files=None):
+        super().__init__(patch_levels, domain_box, refinement_ratio, time, data_files)
+        assert(len(self.levels(time)[0].patches[0].patch_datas.keys()) == 1)
+
+
+
+
+class VectorField(PatchHierarchy):
+
+    # def __init__(self, patch_levels, domain_box, **kwargs):
+    def __init__(self, patch_levels, domain_box, refinement_ratio=2, time=0., data_files=None):
+        # refinement_ratio = kwargs.get("refinement_ratio", 2)
+        # time = kwargs.get("time", .0)
+        # data_files = kwargs.get("data_files", None)
+
+        super().__init__(patch_levels, domain_box, refinement_ratio, time, data_files)
+
+
+    def __mul__(self, right_operand):
+        if isinstance(right_operand, VectorField):
+
+            times_left = list(self.time_hier.keys())
+            times_right = list(right_operand.time_hier.keys())
+
+            patch_levels = self.patch_levels
+            domain_box = self.domain_box
+
+            assert(all(l == r for l, r in zip(times_left, times_right)))
+            for (time_left, time_right) in zip(list(self.time_hier.keys()), list(right_operand.time_hier.keys())):
+                new_patch_level = {}
+
+                assert(all (l == r for l, r in zip(self.levels(time_left).keys(), right_operand.levels(time_right).keys())))
+                for (ilvl_left, lvl_left), (ilvl_right, lvl_right) in zip(self.levels(time_left).items(), right_operand.levels(time_right).items()):
+                    new_patches = {}
+
+                    assert(all (l.id == r.id for l, r in zip(lvl_left.patches, lvl_right.patches)))
+                    for (patch_left, patch_right) in zip(lvl_left.patches, lvl_right.patches):
+                        layout = patch_left.layout
+
+                        if ilvl_left not in new_patches:
+                            new_patches[ilvl_left] = []
+
+                        new_patches[ilvl_left].append(patch_left * patch_right)
```

Comment:
So this is were you're using the `*` operator on patches. I think we should do it on patchdatas.